source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,698 @@
+
+             // Get the deformation type names called for each boundary.
+             std::map<types::boundary_id, std::vector<std::string>> mesh_deformation_boundary_indicators_map
+               = this->get_mesh_deformation_handler().get_active_mesh_deformation_names();
+
+             // Loop over each mesh deformation boundary, and make sure FastScape is only called on the surface.
+             for (std::set<types::boundary_id>::const_iterator p = mesh_deformation_boundary_ids.begin();
+                  p != mesh_deformation_boundary_ids.end(); ++p)
+               {
+                 const std::vector<std::string> names = mesh_deformation_boundary_indicators_map[*p];
+                 for (unsigned int i = 0; i < names.size(); ++i )
+                   {
+                     if (names[i] == "fastscape")
+                       AssertThrow((*p == top_boundary),
+                                   ExcMessage("FastScape can only be called on the surface boundary."));
+                   }
+               }
+
+            // The first entry represents the minimum coordinates of the model domain, the second the model extent.
+            for (unsigned int i=0; i<dim; ++i)
+              {
+                grid_extent[i].first = geometry->get_origin()[i];
+                grid_extent[i].second = geometry->get_extents()[i];
+              }
+
+         nx = repetitions[0] + 1;
+
+         // Size of FastScape cell.
+         dx = (grid_extent[0].second)/( repetitions[0]);
+
+         // FastScape X extent, which is generally ASPECT's extent unless the ghost nodes are used,
+         // in which case 2 cells are added on either side.
+         x_extent = (grid_extent[0].second) ;
+
+        // Sub intervals are 1 less than points.
+        table_intervals[0] = repetitions[0];
+        table_intervals[dim-1] = 1;
+
+        ny = repetitions[1] + 1;
+        dy = (grid_extent[1].second)/( repetitions[1]);
+        table_intervals[1] = repetitions[1];
+        y_extent = (grid_extent[1].second)  ;
+        array_size = nx*ny;
+        
+        
+        
+        
+        
+        
+        
+        
+//
+//      dx = 20000;
+//      dy = dx ;
+//      x_extent = 100e3;
+//      y_extent = 100e3;
+//      nx = 5 ;
+//      ny = 5 ;
+//      array_size = nx*ny;
+//
+//      // Sub intervals are 3 less than points, if including the ghost nodes. Otherwise 1 less.
+//      table_intervals[0] = nx ;
+//      table_intervals[dim-1] = 1;
+//      table_intervals[1] = ny;
+
+//      const GeometryModel::Box<dim> *geometry
+//        = dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model());
+//
+//      // The first entry represents the minimum coordinates of the model domain, the second the model extent.
+//      for (unsigned int i=0; i<dim; ++i)
+//        {
+//          grid_extent[i].first = geometry->get_origin()[i];
+//          grid_extent[i].second = geometry->get_extents()[i];
+//        }
```

Comment:
If you have code that is no longer necessary, just remove it -- or document why you want to keep it.